cirq/study/trial_result.py

```diff
@@ -271,3 +271,25 @@ def __eq__(self, other):
         if not isinstance(other, type(self)):
             return NotImplemented
         return self.data.equals(other.data) and self.params == other.params
+
+    def __add__(self, other: 'TrialResult') -> 'TrialResult':
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        if self.params == other.params and self.measurements.keys(
+        ) == other.measurements.keys():
+            all_measurements: Dict[str, np.ndarray] = {}
+            for key in other.measurements:
```

Comment:
```
def _measurement_shape(self):
    return self.params, {k: v.shape[1] for k, v in self.measurements.items()}
```

```
def __add__(self, other: 'cirq.TrialResult') -> 'cirq.TrialResult':
    if self._result_shape() != other._result_shape():
        raise ValueError(...)
    all_measurements = {
        k: np.append(v1, v2, axis=0) 
        for k, v1, _, v2 in zip(self.measurements.items(), 
                                other.measurements.items())
    }
    return ...
```

Reply:
On second thought, that zip might massively backfire due to ordering.

I wish there was a built-in dictionary pair zip.